### PR TITLE
document optimistic confirmation and slashing roadmap

### PR DESCRIPTION
**Pull request by aeyakovenko**
#### Problem

Optimistic confirmation lacks high level docs.

#### Summary of Changes

Add docs for optimistic confirmation roadmap documentation that covers the slashing and expected security.

Fixes #

### CODE REVIEW

File: docs/src/proposals/optimistic-confirmation-and-slashing.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
+# Optimistic Confirmation and Slashing
+
+Progress on optimistic confirmation can be tracked here
+
+https://github.com/solana-labs/solana/projects/52
+
+At the end of May, the mainnet is moving to 1.1, and TdS is moving
+to 1.2. With 1.2, TdS will behave as if we have 1-block conf as
```
Comment (mvines) [inline docs/src/proposals/optimistic-confirmation-and-slashing.md:8]:
```suggestion
At the end of May, the mainnet-beta is moving to 1.1, and testnet is moving
to 1.2. With 1.2, testnet will behave as if we have 1-block conf as
```

File: docs/src/proposals/optimistic-confirmation-and-slashing.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+tradeoff is that liveness is now reduced by 2X in the worst case.
+If more than 1/3 - 2X of the network is unavailable, the network may
+stall and will resume finalizing blocks after the network recovers.
+So far, we haven’t observed a large unavailability hit on our
+mainnet, cosmos, or tezos. For our network, which is primarily
+composed of high availability systems, this seems unlikely. Currently,
+we have set the threshold percentage to 4.66%, which means that
+23.68% of the nodes have to be available for the network to continue
+finalizing blocks.  For our network, which is primarily composed
+of high availability systems a 23.68 drop in availabilty seems
```
Comment (carllin) [inline docs/src/proposals/optimistic-confirmation-and-slashing.md:41]: 23.68%

File: docs/src/proposals/optimistic-confirmation-and-slashing.md [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+The safety margin is 1/3+X, where X represents the minimum amount
+of stake that will be slashed in case the protocol is violated. The
+tradeoff is that liveness is now reduced by 2X in the worst case.
+If more than 1/3 - 2X of the network is unavailable, the network may
+stall and will resume finalizing blocks after the network recovers.
+So far, we haven’t observed a large unavailability hit on our
+mainnet, cosmos, or tezos. For our network, which is primarily
+composed of high availability systems, this seems unlikely. Currently,
+we have set the threshold percentage to 4.66%, which means that
+23.68% of the nodes have to be available for the network to continue
```
Comment (carllin) [inline docs/src/proposals/optimistic-confirmation-and-slashing.md:39]: 33.333 - 4.6666 * 2 = 24%

Also I think this should read `at most 24% of nodes can be malicious`,  or `at least 76%` of nodes have to be correct.

`24%` is the maximum failure number, not the availability number

File: docs/src/proposals/optimistic-confirmation-and-slashing.md [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+block or that at least 4.66% of the network is violating the protocol.
+
+## How does it work?
+
+The general idea is that validators have to continue voting, following
+their last fork, unless they can construct a proof that their fork
+may not reach finality. The way validators construct this proof is by
+collecting votes for all the other forks, excluding their own. If
+the set of valid votes represents over 1/3+X of the epoch stake weight,
+there is no way for the validators current fork to reach 2/3+ finality.
```
Comment (carllin) [inline docs/src/proposals/optimistic-confirmation-and-slashing.md:20]: I wouldn't say there's `no way`, it's possible, it's possible that some of the `1/3+X` could switch over. It's just that there's also a nonzero possibility that another fork outside of the current one can be finalized, hence why a switch is allowed.

File: docs/src/proposals/optimistic-confirmation-and-slashing.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+Fundamentally, our goal for slashing is to slash 100% in cases where
+the node is maliciously trying to violate safety rules and 0% during
+routine operation. How we aim to achieve that is to first implement
+slashing proofs without any automatic slashing whatsoever.
+
+Right now, for regular consensus, after a safety violation, the
+network will halt. We can analyze the data and figure out who was
+responsible and propose that the stake should be slashed after
+restart. A similar approach will be used with a one-block conf. A
```
Comment (carllin) [inline docs/src/proposals/optimistic-confirmation-and-slashing.md:79]: I would rename all the `one-block conf` to `optimistic confirmation`

**Comment by solana-grimes** [comment]
:broken_heart: Unable to automerge due to CI failure

**Pull request merged**